Improve wxRegKey documentation

PBfordev opened the pull request:
Expand the class description.
Fix and improve the code example.

PBfordev committed:
Improve wxRegKey documentation

Expand the class description.

Fix and improve the code example.

vadz commented on interface/wx/msw/registry.h:
```diff
@@ -34,28 +35,29 @@
+        {
+            wxLogMessage("Subkey name: %s", keyName);
+        }
+        while ( key.GetNextKey(keyName, index) );
```
Considering how rare `do...while` loops are, I'm not sure it's worth using them in an example. Would there be any problem with doing
```cpp
for ( bool cont = key.GetFirstKey(keyName, index); cont; cont = key.GetNextKey(keyName, index) )
{
    ...
}
```
instead?

PBfordev replied:
The original reason why I decided to change the enumeration code was that it used literal `1` where a reference to a `long` (index) was expected, making the code both not only incorrect but also uncompilable.

When I was at it, I changed the algorithm to use what I thought was the usual implementation of the findfirst/findnext API idiom - with a condition check at the end of the loop. But if you find the implementation using `for` to be better, I am going to change the code accordingly.

BTW, I have noticed that headers of some files still have your e-mail address at dptmaths.ens-cachan.fr but I guess that’s preserved for historical reasons and unlike the wxwindows.org one does not need to be changed to point to wxwidgets.org.

vadz replied:
Oh, and forgot to answer this one: yes, that email is very old and doesn't work since ~3 decades, but I don't think it's worth changing it, it should be easy enough to find the working one anyhow.

PBfordev committed:
Change the enumeration loop type

Change the loop type used in key enumeration
from "do/while" to more common "for".

vadz commented:
Thanks for the update!
